Mage/src/main/java/mage/cards/repository/CardCriteria.java

```diff
@@ -217,9 +233,26 @@ public void buildQuery(QueryBuilder qb) throws SQLException {
         optimize();
 
         Where where = qb.where();
-        where.eq("nightCard", nightCard);
+        int clausesCount = 0;
+        if (meldCard == null) {
+            if (nightCard) {
+                where.eq("nightCard", nightCard);
+                clausesCount++;
+            } else {
+                // meldCard are nightCard, so this allows searching for them but not other night cards
+                where.or(
```

Comment:
I don't like that multiple conditionals. Must use simple builder like:

![shot_240407_180552](https://github.com/magefree/mage/assets/8344157/df2e8536-60c3-4b74-ba2e-c932449d117a)

All meld cards are night cards too, so all old code must be compatible.

I don't see a reason for complex code here -- you must use that logic in main code, not in sql builder.

Reply:
Alright, then night cards should be extracted with other cards from the sql,
then later on the non-meld should be filtered out. That was my second option.